examples/examples.test.ts

```diff
@@ -0,0 +1,61 @@
+import { makeRunWithLimit } from "http://localhost:5646/@local/deps/run_with_limit.ts"
+import { Buffer, readLines } from "http://localhost:5646/@local/deps/std/io.ts"
+import * as path from "http://localhost:5646/@local/deps/std/path.ts"
+import { writeAll } from "http://localhost:5646/@local/deps/std/streams.ts"
+import { assert } from "http://localhost:5646/@local/deps/std/testing/asserts.ts"
+
+const IS_CI = Deno.env.get("ENV_TYPE_CI") ? true : false
+
+function getModuleDir(importMeta: ImportMeta): string {
+  return path.resolve(path.dirname(path.fromFileUrl(importMeta.url)))
+}
+
+async function pipeThrough(
+  reader: Deno.Reader,
+  writer: Deno.Writer,
+) {
+  const encoder = new TextEncoder()
+  for await (const line of readLines(reader)) {
+    await writeAll(writer, encoder.encode(`${line}\n`))
+  }
+}
+
+const ignoreFile = await Deno.readTextFile(path.join(getModuleDir(import.meta), ".ignore"))
+const ignoredFiles = new Set(ignoreFile.split("\n"))
+
+const exampleFileNames = Array.from(Deno.readDirSync(getModuleDir(import.meta)))
+  .filter((e) => e.name.match(/^.*\.ts$/g) && e.isFile && !ignoredFiles.has(e.name))
+  .map((f) => f.name)
+
+const { runWithLimit } = makeRunWithLimit<void>(navigator.hardwareConcurrency)
+
+const exampleTasks = IS_CI
+  ? exampleFileNames.map((fileName) =>
+    runWithLimit(async () => {
+      const t = Deno.run({
+        cmd: ["deno", "task", "run", `${getModuleDir(import.meta)}/${fileName}`],
+        stdout: "piped",
+        stderr: "piped",
+      })
+
+      const out = new Buffer()
+      await Promise.all([pipeThrough(t.stdout, out), pipeThrough(t.stderr, out)])
+
+      const status = await t.status()
+
+      t.close()
+
+      return Deno.test({
+        name: `examples/${fileName}`,
+        async fn() {
+          if (!status.success) {
+            await pipeThrough(out, Deno.stderr)
```

Comment:
This should happen synchronously through a `console.log`, ideally prefixed with the example name, e.g.:
```
[examples/file.ts]
output line 1
output line 2
...
```

Reply:
each example wil have the `task run` prefixed with it so should be fine. 

example:

```
Task run deno run -A "/workspaces/capi/examples/block_author.ts"
14DE8GdKnNvgoXCLFq62ZjNz2zsGqnxXBsRMwNpiPip2JSFJ
error: Uncaught (in promise) Error: aaa
throw new Error("aaa")
      ^
    at file:///workspaces/capi/examples/block_author.ts:43:7
```